### ISSUE TITLE
Android Studio presence changes flutter toolchain 

### ISSUE THREAD

**Issue by VictorXLR** [flutter/flutter#98072]
Issue: Flutter uses the version of java bundled with android studio by default instead of the base system version, this is a problem because the android studio java (11) doesn't support `HmacPBESHA256 ` an algorithm used to sign builds when deploying for android. 

The only way to fix this currently is to remove android studio and develop with an alternative like `visual studio code` which is not sufficient as android studio is a much better IDE. 

## Steps to Reproduce
 - Create a flutter project and follow the steps here to signing the app https://docs.flutter.dev/deployment/android#signing-the-app
 - Note: This issue originally started when trying to develop on a windows and mac machines where it wouldnt build development or release builds of our app due to the lack of the HmacPBESHA256 algorithm on JAVA 11.  When running `flutter build apk` or `flutter build appbundle` or `flutter install` on a physical device. the app would generate a stack error of this. 
```
Execution failed for task ':app:packageRelease'.
> A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
   > com.android.ide.common.signing.KeytoolException: Failed to read key upload from store "C:\Users\*\StudioProjects\*\*\app\upload-keystore.jks": Integrity check failed: java.security.NoSuchAlgorithmException: Algorithm HmacPBESHA256 not available
```
I followed the suggestion on stack overflow and changed the version of Java to Java 17, from the project structure page 
![image](https://user-images.githubusercontent.com/22397950/153084352-79d1f9a8-1636-4b4f-b9c4-1b3720bcf859.png)
and also from `gradle -version`, it says that it uses java 17
![image](https://user-images.githubusercontent.com/22397950/153084590-59e2f2fc-fb36-409b-92e4-5ddf6143b9c8.png)
but i am unable to run `flutter build apk` and `flutter build appbundle` 
due to the same error 
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:packageRelease'.
> A failure occurred while executing com.android.build.gradle.tasks.PackageAndroidArtifact$IncrementalSplitterRunnable
   > com.android.ide.common.signing.KeytoolException: Failed to read key upload from store "C:\Users\**\StudioProjects\**\android\app\upload-keystore.jks": Integrity check failed: java.security.NoSuchAlgorithmException: Algorithm HmacPBESHA256 not available

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 34s
Running Gradle task 'assembleRelease'...                           95.2s
Gradle task assembleRelease failed with exit code 1
```
Flutter Doctor (note: android toolchain, uses java 11)
![image](https://user-images.githubusercontent.com/22397950/153085594-34ad1c6c-7832-45d3-95be-3b7c92feecd3.png)

I updated my enviroment variables for `JAVA_HOME` and `STUDIO_JDK` to my base java version and its unable to work 
```
PS C:\Users\Victor Popoola> $Env:STUDIO_JDK
C:\Program Files\Java\jdk-17.0.1
PS C:\Users\Victor Popoola> $Env:JAVA_HOME
C:\Program Files\Java\jdk-17.0.1
PS C:\Users\Victor Popoola>
```
so i found the solution by:  **Uninstall Android Studio**

Flutter doctor - without android studio installed, (it uses java 17 instead of 11)
![image](https://user-images.githubusercontent.com/22397950/153087624-553025c7-1b44-4456-b523-a9632fabb564.png)

after uninstalling android studio, I am now able to run and build my app with the system version of java. 
I am unable to build my app for android or install on host devices due to the fact that Flutter will choose the java version bundled with android studio over the one installed on the base system. i have had this issue with mac and windows machine and the current solution for this seems to be uninstalling android studio.

if you reinstall android studio, the same issue comes back and you are unable to get any work done. Unfortunately android studio is a much better ide for flutter than visual studio code. so i am providing this issue with the hopes that it gets fixed. 

<details>
<summary>Code sample</summary>

- create a simple flutter project
- follow the guide here to sign the app https://docs.flutter.dev/deployment/android#signing-the-app
- modify `android/app/build.gradle` to utilize the keytool 
```
signingConfigs {
        release {
            keyAlias keystoreProperties['keyAlias']
            keyPassword keystoreProperties['keyPassword']
            storeFile keystoreProperties['storeFile'] ? file(keystoreProperties['storeFile']) : null
            storePassword keystoreProperties['storePassword']
        }
    }
    buildTypes {
        release {
            minifyEnabled true
            shrinkResources true
            signingConfig signingConfigs.release
        }
    }
```
- try and build the app.
</details>

**Comment by darshankawar** [comment]
@VictorXLR 
Check [this](https://stackoverflow.com/questions/67631927/error-building-aab-flutter-android-integrity-check-failed-java-security-n) SO link and underlying comments and see if any of them help to resolve this issue.

Also, there's similar open issue reported in flutter/website repo, since the instructions were followed using the steps mentioned in it: https://github.com/flutter/website/issues/5871. Check underlying comments from it to see if they help.

Closing from here, since it may be related to Android Studio itself and not with Flutter framework. If you disagree, write in comments and I'll reopen it.

**Comment by VictorXLR** [comment]
Hi,

I have been on that SO page multiple times and setting my java version in gradle.properties to my system path didnt help,
I already have my keytool signed with the `-storetype jks` arguement too. I use `jetbrains toolbox` on my mac and desktop and i am running on the latest version of android studio `bumblebee patch 1` 

 the problem isn't with android studio but with flutter as `flutter doctor -v` will use the version of java bundled with android studio instead of the base system one. 

The only way i am able to fix it is for me to uninstall android studio entirely, no matter how many times i try to build my app bundle with android studio installed, it never worked until i removed android studio and then `flutter doctor -v` would use the system base java version over the one bundled with android studio.

Even with android studio closed and from the terminal and with `./gradlew -version` showing java 17, it will still use `java 11` from android studio over anything else. I updated my environment variables in my initial issue message and it didn't work.


I provided screenshots of my flutter doctor with android studio installed and not installed in my initial message to show the difference
